addons/edi/models/edi_document.py

```diff
@@ -260,6 +261,52 @@ class EdiDocument(models.Model):
 
     fail_fast = fields.Boolean(related="doc_type_id.fail_fast", readonly=True)
 
+    # Processing statistics
+    waiting_time_start = fields.Datetime(string="Waiting start", readonly=True)
+    waiting_time_end   = fields.Datetime(string="Waiting end", readonly=True)
+    waiting_time_total = fields.Float(
+        string="Waiting total time", compute="_compute_waiting_time",
+        store=True, readonly=True
+    )
+    prepare_time_start = fields.Datetime(string="Prepare start", readonly=True)
+    prepare_time_end   = fields.Datetime(string="Prepare end", readonly=True)
+    prepare_time_total = fields.Float(
+        string="Prepare total time", compute="_compute_prepare_time",
+        store=True, readonly=True
+    )
+    execute_time_start = fields.Datetime(string="Execution start", readonly=True)
+    execute_time_end   = fields.Datetime(string="Execution end", readonly=True)
+    execute_time_total = fields.Float(
+        string="Execution total time", compute="_compute_execute_time",
+        store=True, readonly=True
+    )
+
+    received_time = fields.Datetime(
+        string="Received at", default=fields.Datetime.now(), readonly=True
+    )
+    completed_time = fields.Datetime(string="Completed at", readonly=True)
+    processing_time_total = fields.Float(
+        string="Total processing time",
+        compute="_compute_processing_time",
+        store=True,
+        readonly=True,
+        help="Total processing time since receiving",
+    )
+    total_records = fields.Integer(
```

Comment:
Question: What is the difference between total_records and total_records processed?

Reply:
It might be an idea just to log the total_records_processed once the doc has executed and remove the total_records field.

Reply:
If only the state is the difference, i think dont need 2 fields. We can understand that are total processed if edi doc is in done state

Reply:
Do we need to save the information we log on server logs, to be visible from the users while is being executed so the know they are executed these records out of total records. Practically similar information with server logs. maybe it is and didnt understand which part of code does this

Reply:
`total_records_processed` is the number of successfully processed records.

Reply:
total_records is the number of total records i suppose :)

Reply:
For document types set as not fail fast; it might be possible that the total records and total processed records can differ, though (if some records fail, for instance).